iam.tf

```diff
@@ -0,0 +1,84 @@
+locals {
+  parsed_state_files = [
+    for url in var.state_files : {
+      bucket = regex("s3://([^/]+)/.*", url)[0]
+      prefix = regex("s3://[^/]+/(.*)", url)[0]
```

Comment:
Does this let you also specify prefix paths to multiple state files instead of/as well as the absolute paths of individual state files?